Deployment Guides/Provisioning-Islandora-on-Ubuntu-16.md

```diff
@@ -0,0 +1,1807 @@
+    NUMBER_OF_GSEARCH_THREADS="2"          #do one gsearch thread per cpu might want to skip multithreading if you only have 1 cpu
+    GSEARCH_UPDATER_NAMES="FgsUpdaters FgsUpdater1"
+    DRUPAL_FILTER_URL="https://github.com/Islandora/islandora_drupal_filter/releases/download/v7.1.3/fcrepo-drupalauthfilter-3.8.1.jar"
+    # NOTE as above, chose the pair following, depending on the version of Solr you are installing
+    #SOLR_URL="http://archive.apache.org/dist/lucene/solr/4.2.0/solr-4.2.0.tgz"
+    #SOLR_NAME="solr-4.2.0"
+    #SOLR_URL="http://archive.apache.org/dist/lucene/solr/4.10.4/solr-4.10.4.tgz"
+    #SOLR_NAME="solr-4.10.4"
+    SOLR_DEFAULT_CORE_PATH="collection1"
+
+    Q1="CREATE DATABASE IF NOT EXISTS $DRUPAL_DB_NAME CHARACTER SET utf8 COLLATE utf8_bin;"
+    Q2="GRANT ALL ON $DRUPAL_DB_NAME.* TO $DRUPAL_DB_USER@localhost IDENTIFIED BY '$DRUPAL_DB_PASS';"
+    Q3="CREATE DATABASE IF NOT EXISTS $FEDORA_DB_NAME CHARACTER SET utf8 COLLATE utf8_bin;"
+    Q4="GRANT ALL ON $FEDORA_DB_NAME.* TO $FEDORA_DB_USER@localhost IDENTIFIED BY '$FEDORA_DB_PASS';"
+    Q5="FLUSH PRIVILEGES;"
+    SQL="${Q1}${Q2}${Q3}${Q4}${Q5}"
+
+    APACHE_DEFAULT_SITE="000-default"
+    # NOTE: We are using php 7.1, and APC is now deprecated, in favour of Zend Opcache, so the following needs to be dealt with at a later date
+    APC_CONFIG_FILE="/etc/php/7.1/mods-available/apcu.ini"
+    APACHE_USER="www-data"
+    APACHE_SERVICE="apache2"
+    OS_DEFAULT_DOCUMENTROOT="/var/www"
+    CRON_SPOOL_DIR="/var/spool/cron/crontabs"
+    #Resolvable hostname.
+    DNS_HOSTNAME="yourhostname.local"
+```
+
+Source environment _(Please note: this must be redone if you close your shell mid-install)_:
+```
+chmod +x ~/islandora-install.properties
+
+. ~/islandora-install.properties
+```
+
+### Operating System <a id="operating-system"></a>
+1. Ensure latest core OS updates: `apt-get update && apt-get upgrade && apt-get --with-new-pkgs upgrade`. The latter will ensure you get the latest kernel patches. A reboot will probably be needed at this point.
+
+2. Add repo to easily obtain oracle java installer: `apt-get -y install python-software-properties && add-apt-repository -y ppa:webupd8team/java && apt-get -y update`
+
+3. Add optional multiverse repos in `/etc/apt/sources.list`. Uncomment each line in this file that ends in multiverse.
+
+4. Add optional repo to get PHP7.1 material: `add-apt-repository -y ppa:ondrej/php && apt-get update -y`
+
+## Software Dependencies <a id="package-installation"></a>
+
+### Software Dependencies Installed by Apt-Get Package Manager <a id="from-apt-get"></a>
+Accept license agreement and enter root password when asked.
+
+Note: we use php7.1, while both 7.0 and 7.2 are available. 7.0 is too close to end of life, and 7.2 produces difficult to manage 'deprectated' errors in Drupal 7 (the 'each()' function is deprecated as of 7.2). You can go to 7.2, but setting 'error_reporting' to exclude deprecations doesn't seem to do anything.
```

Comment:
Ubuntu will actually continue to backport security fixes for 7.0 until the end of the LTS just like they currently do for php 5.5.9.

Reply:
I'll add a note.